asn1ate/asn2quickder.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/python
+#
+# asn2quickder -- Generate header files for C for use with Quick `n' Easy DER
+#
+# This program owes a lot to asn1ate, which was built to generate pyasn1
+# classes, but which was so well-written that it could be extended with a
+# code generator for Quick DER.
+#
+# Much of the code below is diagonally inspired on the pyasn1 backend, so
+# a very big thank you to Schneider Electric Buildings AB for helping to
+# make this program possible!
+#
+# From: Rick van Rein <rick@openfortress.nl>
+
+
+import sys
+import os.path
+
+from asn1ate import parser
+from asn1ate.sema import * 
+
+
+def toCsym (name):
+	"""Replace unsupported characters in ASN.1 symbol names"""
+	return str (name).replace (' ', '').replace ('-', '_')
+
+
+class QuickDERgen ():
+	"""Generate the C header files for Quick DER, a.k.a. Quick and Easy DER.
+
+	   There are two things that are generated for each of the ASN.1 syntax
+	   declaration symbol of a unit:
+
+	   #define DER_PACK_unit_SyntaxDeclSym \
+			   DER_PACK_ENTER | ..., \
+			   ... \
+			   DER_PACK_LEAVE, \
+			   DER_PACK_END
+
+	   this is a walking path for the der_pack() and der_unpack() instructions.
+	   In addition, there will be a struct for each of the symbols:
+
+	   struct unit_SyntaxDeclSym_ovly {
+		   dercursor field1;
+		   dercursor field2;
+		   struct unit_EmbeddedSym_ovly field3;
+		   dercursor field4;
+	   };
+
+	   The unit prefix will be set to the filename of the module, usually
+	   something like rfc5280 when the parsed file is rfc5280.asn1 and the
+	   output is then written to rfc5280.h for easy inclusion by the C code.
+	"""
+
+	def __init__ (ik, semamod, outfn, refmods):
+		ik.semamod = semamod
+		ik.refmods = refmods
+		if outfn [-2:] == '.h':
+			raise Exception ('File cannot overwrite itself -- use another extension than .h for input files')
+		ik.unit = toCsym (outfn.rsplit ('.', 1) [0])
+		ik.outfile = open (ik.unit + '.h', 'w')
+		ik.wout = ik.outfile.write
+		# Setup function maps
+		ik.ovly_funmap = {
```

Comment:
I'm guessing `ovly` is short for `overlay` -- would it be clearer to spell it out? Or is `ovly` a strong concept in QuickDER so that it makes sense to reflect it in this code generator?